{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# A03 Key\n",
    "\n",
    "## Problem 1\n",
    "\n",
    "What is one of the primary challenges in de novo gene assembly?\n",
    "\n",
    "**A.** The uniformity of genetic sequences across different organisms, making it difficult to distinguish between species.\n",
    "\n",
    "**B.** The presence of repetitive sequences within the genome that can complicate the assembly process.\n",
    "\n",
    "**C.** The overabundance of RNA sequences in DNA sequencing data, interfering with genome assembly.\n",
    "\n",
    "**D.** The rapid degradation of DNA samples during the sequencing process, resulting in incomplete data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "problem_1_answer = \"B\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Explanation\n",
    "\n",
    "The uniformity of genetic sequences across different organisms makes it challenging to distinguish between species.\n",
    "\n",
    "> This option is incorrect because de novo gene assembly primarily involves assembling sequences from a single organism's genome rather than distinguishing between species.\n",
    "> The challenge in de novo assembly is not about the uniformity across species but rather about accurately piecing together the genome of a single organism.\n",
    "\n",
    "**The presence of repetitive sequences within the genome can complicate the assembly process.**\n",
    "\n",
    "> This option is correct.\n",
    "> One of the primary challenges in de novo gene assembly is the presence of repetitive sequences within the genome.\n",
    "> These repeats can confuse assembly algorithms, making it difficult to accurately reconstruct the original genome from short sequencing reads.\n",
    "> Repetitive sequences can lead to assembly errors, such as collapses or expansions of repeat regions.\n",
    "\n",
    "The overabundance of RNA sequences in DNA sequencing data, interfering with genome assembly.\n",
    "\n",
    "> This option is incorrect because DNA sequencing data, especially in the context of de novo genome assembly, focuses on DNA rather than RNA. While RNA-seq data can be used for transcriptome assembly, the primary challenge in de novo DNA assembly is not the presence of RNA sequences but dealing with the DNA sequences themselves.\n",
    "> Contamination and quality control are concerns in sequencing data.\n",
    "> Still, the overabundance of RNA sequences is not a direct challenge for de novo DNA assembly.\n",
    "\n",
    "The rapid degradation of DNA samples during the sequencing process results in incomplete data.\n",
    "\n",
    "> This option is incorrect as a primary challenge. While DNA degradation can be a problem in sequencing, modern sequencing technologies and protocols are designed to minimize the impact of degradation.\n",
    "> The primary challenge in de novo assembly is not the degradation of DNA samples but rather computational and algorithmic challenges related to accurately reconstructing genomes from short, error-prone reads."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Problem 2\n",
    "\n",
    "Which of the following statements accurately reflects a challenge associated with de novo assembly, especially when compared to reference assembly?\n",
    "\n",
    "**A.** De novo assembly is less effective at detecting single nucleotide polymorphisms (SNPs).\n",
    "\n",
    "**B.** De novo assembly cannot be used to study organisms with large or complex genomes, such as humans or plants.\n",
    "\n",
    "**C.** De novo assembly typically requires higher sequencing depth to achieve comparable accuracy and completeness to reference assembly.\n",
    "\n",
    "**D.** Reference assembly is preferred for metagenomic studies due to its superior ability to reconstruct genomes from mixed samples.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "problem_2_answer = \"C\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Explanation\n",
    "\n",
    "De novo assembly is less effective at detecting single nucleotide polymorphisms (SNPs).\n",
    "\n",
    "> This statement is somewhat misleading.\n",
    "> While de novo assembly does not rely on a reference genome for SNP identification, it can still be used to detect SNPs once the assembly is complete.\n",
    "> The challenge here is not the effectiveness in detecting SNPs per se but rather the complexity of assembling a genome without a reference.\n",
    "> Once assembled, the novel genome can serve as a reference for SNP detection through comparison with other sequences.\n",
    "> Thus, the main challenge with de novo assembly in the context of SNPs is not an inherent inability to detect them but rather the initial difficulty of assembling the genome to a high enough quality that can then be used for such analyses.\n",
    "\n",
    "De novo assembly cannot be used to study organisms with large or complex genomes, such as humans or plants.\n",
    "\n",
    "> This statement is incorrect.\n",
    "> De novo assembly can and has been used to study organisms with large and complex genomes.\n",
    "> The challenge is not that it cannot be used but that it requires substantial computational resources and often a higher sequencing depth to manage the complexity and size of these genomes.\n",
    "> Advances in sequencing technologies and assembly algorithms have increasingly made it possible to tackle large genomes de novo, although with varying degrees of difficulty and resource requirements.\n",
    "\n",
    "De novo assembly typically requires higher sequencing depth to achieve comparable accuracy and completeness to reference assembly.\n",
    "\n",
    "> This statement needs to be more general and more accurate.\n",
    "> The preference for reference assembly or de novo assembly in metagenomic studies depends on the specific goals of the study and the availability of reference genomes.\n",
    "> De novo assembly can be particularly valuable in metagenomics for discovering novel organisms without reference genome.\n",
    "> However, reference assembly can be more straightforward and less computationally intensive when references are available, and the goal is to identify known organisms within a sample.\n",
    "> Both approaches have their place in metagenomics, with de novo assembly playing a critical role in exploring the genetic material of uncultured or unknown microbes.\n",
    "\n",
    "**Reference assembly is preferred for metagenomic studies due to its superior ability to reconstruct genomes from mixed samples.**\n",
    "\n",
    "> Without a reference genome to guide the alignment of sequencing reads, de novo assembly relies on overlapping sequences to assemble the genome or transcriptome from scratch.\n",
    "> This process is more susceptible to errors, especially in repetitive regions or areas with low complexity. It is necessary to have higher sequencing coverage to ensure that all regions are accurately assembled.\n",
    "> The increased sequencing depth helps compensate for these challenges by providing more data for the assembly algorithm, improving the chances of correctly piecing together the genome.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Problem 3\n",
    "\n",
    "**Points:** 5\n",
    "\n",
    "Imagine you are working on a simplified de novo gene assembly project where you are given a set of short DNA reads.\n",
    "Your task is to determine the shortest possible DNA sequence that contains all the given reads as subsequences. Each read is represented by a string consisting of characters A, T, C, and G, which stand for the nucleotides adenine, thymine, cytosine, and guanine, respectively.\n",
    "\n",
    "Given the following reads:\n",
    "\n",
    "-   Read 1: ATCG\n",
    "-   Read 2: CGTT\n",
    "-   Read 3: GATC\n",
    "-   Read 4: TCGA\n",
    "\n",
    "Determine the overlap between each pair of reads and combine them all into one contig.\n",
    "Start with read 1, merge with the highest overlap, then repeat until all reads are used.\n",
    "\n",
    "Put your final sequence in `problem_3_answer`; for example, `problem_3_answer = \"AAAAATTTTTCCGGC\"`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "problem_3_answer = \"TCGATCGTT\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Explanation\n",
    "\n",
    "In the normal greedy algorithm, you have to try every possible merge to find the largest overlap.\n",
    "To make it easier, I gave you the instruction to start with a merge involving read 1.\n",
    "\n",
    "**Merge 1**\n",
    "\n",
    "Try to merge `ATCG` with `CGTT`.\n",
    "\n",
    "```text\n",
    "ATCG--\n",
    "--CGTT\n",
    "```\n",
    "\n",
    "or \n",
    "\n",
    "```text\n",
    "----ATCG\n",
    "CGTT----\n",
    "```\n",
    "\n",
    "Maximum overlap of 2.\n",
    "\n",
    "**Merge 2**\n",
    "\n",
    "Try to merge `ATCG` with `GATC`.\n",
    "\n",
    "```text\n",
    "ATCG---\n",
    "---GATC\n",
    "```\n",
    "\n",
    "or \n",
    "\n",
    "```text\n",
    "-ATCG\n",
    "GATC-\n",
    "```\n",
    "\n",
    "Maximum overlap of 3.\n",
    "\n",
    "**Merge 3**\n",
    "\n",
    "Try to merge `ATCG` with `TCGA`.\n",
    "\n",
    "```text\n",
    "ATCG-\n",
    "-TCGA\n",
    "```\n",
    "\n",
    "or\n",
    "\n",
    "```text\n",
    "---ATCG\n",
    "TCGA---\n",
    "```\n",
    "\n",
    "Maximum overlap of 3.\n",
    "\n",
    "**(Bonus) Merge 4**\n",
    "\n",
    "Try to merge `CGTT` with `GATC`.\n",
    "\n",
    "```text\n",
    "CGTT----\n",
    "----GATC\n",
    "```\n",
    "\n",
    "or\n",
    "\n",
    "```text\n",
    "---CGTT\n",
    "GATC---\n",
    "```\n",
    "\n",
    "Maximum overlap of 1.CGTT\n",
    "Try to merge `CGTT` with `TCGA`.\n",
    "\n",
    "```text\n",
    "CGTT---\n",
    "---TCGA\n",
    "```\n",
    "\n",
    "or\n",
    "\n",
    "```text\n",
    "----CGTT\n",
    "TCGA----\n",
    "```\n",
    "\n",
    "Maximum overlap of 1.\n",
    "\n",
    "**(Bonus) Merge 6**\n",
    "\n",
    "Try to merge `GATC` with `TCGA`.\n",
    "\n",
    "```text\n",
    "GATC\n",
    "--TCGA\n",
    "```\n",
    "\n",
    "or\n",
    "\n",
    "```text\n",
    "--GATC\n",
    "TCGA\n",
    "```\n",
    "\n",
    "Maximum overlap of 2."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Since we have two equal probable merges `GATC` + `ATCG` and `ATCG` + `TCGA`, we have to consider both options.\n",
    "\n",
    "**Merge 2.1**\n",
    "\n",
    "Try to merge `GATCG` with `CGTT`.\n",
    "\n",
    "```text\n",
    "GATCG--\n",
    "---CGTT\n",
    "```\n",
    "\n",
    "or\n",
    "\n",
    "```text\n",
    "----GATCG\n",
    "CGTT-----\n",
    "```\n",
    "\n",
    "Maximum overlap of 2.\n",
    "\n",
    "**Merge 2.2**\n",
    "\n",
    "Try to merge `GATCG` with `TCGA`.\n",
    "\n",
    "```text\n",
    "GATCG-\n",
    "--TCGA\n",
    "```\n",
    "\n",
    "or\n",
    "\n",
    "```text\n",
    "--GATCG\n",
    "TCGA---\n",
    "```\n",
    "\n",
    "Maximum overlap of 3.\n",
    "This is a merge with the most overlap, so we proceed with : `TCGATCG`.\n",
    "\n",
    "**Merge 2.2.1**\n",
    "\n",
    "There is only one fragment left, so let's try to merge it.\n",
    "\n",
    "```text\n",
    "TCGATCG--\n",
    "-----CGTT\n",
    "```\n",
    "\n",
    "or\n",
    "\n",
    "```text\n",
    "---TCGATCG\n",
    "CGTT------\n",
    "```\n",
    "\n",
    "We have a maximum overlap of 2, which would result in **`TCGATCGTT`**."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "What happens if we made the other move?\n",
    "\n",
    "**Merge 3.1**\n",
    "\n",
    "Try to merge `ATCGA` with `CGTT`.\n",
    "\n",
    "```text\n",
    "ATCGA----\n",
    "-----CGTT\n",
    "```\n",
    "\n",
    "or\n",
    "\n",
    "```text\n",
    "----ATCGA\n",
    "CGTT-----\n",
    "```\n",
    "\n",
    "Maximum overlap of 0.\n",
    "\n",
    "**Merge 3.2**\n",
    "\n",
    "Try to merge `ATCGA` with `GATC`.\n",
    "\n",
    "```text\n",
    "ATCGA----\n",
    "---GATC--\n",
    "```\n",
    "\n",
    "or\n",
    "\n",
    "```text\n",
    "-ATCGA\n",
    "GATC--\n",
    "```\n",
    "\n",
    "Maximum overlap of 3.\n",
    "We proceed with this merge which would result in `GATCGA`.\n",
    "\n",
    "**Merge 3.2.1**\n",
    "\n",
    "Try to merge `GATCGA` with `CGTT`.\n",
    "\n",
    "```text\n",
    "GATCGA----\n",
    "------CGTT\n",
    "```\n",
    "\n",
    "or\n",
    "\n",
    "```text\n",
    "----GATCGA\n",
    "CGTT------\n",
    "```\n",
    "\n",
    "There is no overlap, so we do not have a valid merge; thus, the answer is Merge 2.2.1."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Problem 4\n",
    "\n",
    "You are given the following sequencing error probabilities from an illumina run:\n",
    "\n",
    "-   0.0398%\n",
    "-   0.0398%\n",
    "-   0.0398%\n",
    "-   0.0398%\n",
    "-   0.0398%\n",
    "-   0.0158%\n",
    "-   0.0316%\n",
    "-   0.0158%\n",
    "-   0.0251%\n",
    "-   0.02%\n",
    "\n",
    "What would the ASCII characters be in the FASTQ file?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "problem_4_answer = \"CCCCCGDGEF\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The sequencing error probabilities provided correspond to the error rates observed in an Illumina sequencing run.\n",
    "In FASTQ format, quality scores are encoded as ASCII characters, with each character representing a Phred quality score.\n",
    "The Phred quality score, Q, is related to the error probability, P, by the formula:\n",
    "\n",
    "$$\n",
    "Q = âˆ’10 \\log_{10}(P)\n",
    "$$\n",
    "where P is the error probability (e.g., 0.0398% or 0.000398 in decimal form).\n",
    "\n",
    "Let's calculate the Phred quality scores for each of the given error probabilities and then convert these scores to their corresponding ASCII characters.\n",
    "In the FASTQ format, a Phred score of Q is encoded by an ASCII character with a value of Q+33, following the Sanger format.\n",
    "\n",
    "Let's perform these calculations.\n",
    "\n",
    "The ASCII characters corresponding to the given sequencing error probabilities in the FASTQ file would be:\n",
    "\n",
    "-   C for the error probabilities of 0.0398%\n",
    "-   G for the error probabilities of 0.0158%\n",
    "-   D for the error probability of 0.0316%\n",
    "-   E for the error probability of 0.0251%\n",
    "-   F for the error probability of 0.02%\n",
    "\n",
    "These characters represent the encoded Phred quality scores, which reflect the reliability of each base call in the sequence data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CCCCCGDGEF\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "\n",
    "# Given error probabilities in percentage\n",
    "error_percentages = [\n",
    "    0.0398,\n",
    "    0.0398,\n",
    "    0.0398,\n",
    "    0.0398,\n",
    "    0.0398,\n",
    "    0.0158,\n",
    "    0.0316,\n",
    "    0.0158,\n",
    "    0.0251,\n",
    "    0.0200,\n",
    "]\n",
    "\n",
    "# Convert percentages to decimal probabilities\n",
    "error_probabilities = np.array(error_percentages) / 100\n",
    "\n",
    "# Calculate Phred quality scores\n",
    "Q_scores = -10 * np.log10(error_probabilities)\n",
    "\n",
    "# Convert Phred scores to ASCII characters\n",
    "ascii_characters = \"\".join([chr(int(round(q)) + 33) for q in Q_scores])\n",
    "\n",
    "print(ascii_characters)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "biosc1540-2024s-dev",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
