# Antimicrobial resistance

Antibiotics, hailed as revolutionary in combating bacterial infections, have transformed medicine and saved countless lives.
Their applications extend beyond medicine.
They are widely used in animal husbandry and production for preventive purposes in various countries.
However, the extensive and sometimes inappropriate use of antibiotics has given rise to antimicrobial resistance (AMR).
In this phenomenon, microorganisms, including bacteria, viruses, fungi, and parasites, adapt and thrive despite exposure to drugs designed to eliminate them.

AMR poses significant challenges, making infections caused by resistant organisms challenging to treat and increasing the risk of severe illness and death.
Various antimicrobial agents aim to control microbial growth, such as antibiotics, antifungals, antivirals, disinfectants, and food preservatives.
The global surge in antibiotic use, particularly in developing nations, provides bacteria ample opportunities to develop AMR, leading to higher morbidity and mortality rates.
The incidence of antimicrobial-resistant bacterial infections has reached alarming levels in the 21st century, posing a silent pandemic that demands urgent intervention to safeguard global public health.

AMR's evolution and spread result from numerous interconnected factors related to healthcare, agriculture, pharmaceuticals, waste management, trade, and finance.
Recognized as one of the most complex public health concerns worldwide, AMR is exacerbated by the rapid spread of "superbugs," microorganisms resistant to most known antimicrobials.

The World Health Organization (WHO) identifies AMR as a top-three public health threat, ranking third as a cause of death after cardiovascular diseases.
A study estimates that in 2019, 1.27 million deaths were attributable to antimicrobial-resistant infections, with a projected increase to 10 million per year by 2050.
The misuse of antibiotics, coupled with their inappropriate dosage and duration, has led to the emergence of resistant bacteria.
The unscientific use of antimicrobials in livestock feed in developing countries further contributes to the spread of AMR.

Despite the growing demand for novel antimicrobials to combat life-threatening infections, pharmaceutical giants have reduced their focus on antibiotic discovery since the 1980s, leading to a need for more new developments.
The lack of significant discoveries underscores the importance of preserving the efficacy of existing antimicrobials by avoiding unnecessary consumption.

## Types of resistance

Antibiotic resistance arises in bacteria as an adaptive response to therapeutic antibiotics.
Clinically, targeted pathogens' initial susceptibility to a newly introduced antibiotic diminishes over time due to prolonged use, leading to bacterial resistance.
Evolutionarily, bacteria modify their response to antibiotics through chromosomal gene mutations or the acquisition of foreign DNA via horizontal gene transfer (HGT), which carries resistance determinants.
Mutations typically involve genes related to antibiotic targets, transporters, and regulators that control transporter expression, resulting in antibiotic resistance through mechanisms such as antibiotic-modifying enzymes and multidrug efflux pumps.

Compelling evidence suggests that commensal or environmental bacteria are the origin of antibiotic-resistance genes transmitted to human pathogens through HGT.
Environmental microorganisms naturally produce various antibiotics, necessitating the presence of antibiotic-resistant genes to protect them from their antibiotic effects.

### Intrinsic

Intrinsic resistance is an inherent natural capacity of bacteria to resist certain antibiotics due to the presence of their chromosomal genes.
Efflux pumps and reduced permeability are common mechanisms involved in intrinsic resistance.

### Acquired

Acquired resistance results from the evolutionary process of a previously sensitive bacterium gaining resistance through chromosomal gene mutation or acquiring new genetic material via HGT.
Conjugation, transformation, and transposition are key mechanisms for HGT, with acquired resistance often transmitted through a conjugated plasmid, either temporarily or permanently.

### Adaptive

Adaptive resistance is a conditional phenotype influenced by environmental changes.
It depends on the duration and strength of selection pressure and is typically temporary.
Bacterial growth influenced by subinhibitory antibiotic concentrations and specific environmental signals can lead to adaptive resistance in humans and livestock.
Unlike intrinsic and acquired resistance, adaptive resistance is transient.
Generally, it reverts to the original state upon removal of the inducing signals.
While the biological processes involved are not fully understood, factors such as high mutation rates, gene amplification, efflux pumps, biofilm formation, epigenetic inheritance, population structure, and heterogeneity are considered potential explanations for its development.

## Transmission

The spread and acquisition of antimicrobial resistance (AMR) primarily occur through human-to-human interactions within and outside healthcare facilities.
Reservoirs for AMR include humans, animals, water, and the environment, with the potential for antimicrobial-resistance genes to be transmitted among these reservoirs.
Notably, there are significant variations in transmission routes among bacterial species and resistance elements.

Specific sources act as hotspots for the facilitated transmission of antimicrobial-resistant bacteria.
Examples include wastewater and sludge from urban wastewater treatment plants and natural fertilizers like pig slurry, cow manure, and poultry farming fertilizer.
Additionally, the direct route of acquiring antimicrobial resistance from animals involves animal feeds treated with antibiotics and subsequent transfer to humans through the consumption of these animals.
Common transmission routes include ingesting fecal-contaminated food or water and direct contact between animals and humans.

## Mechanism

Antimicrobials and bacteria share a common ecological space, with bacteria evolving defenses to counteract the detrimental impact of antibiotic molecules.
Antibiotics typically target four essential components within a bacterial cell: the cell wall, cell membrane, protein synthesis, and nucleic acid synthesis.
Key mechanisms contributing to antimicrobial resistance involve restricting drug entry, modifying drug targets, deactivating drugs, and enhancing the efflux of active drugs.

!!! quote "**Figure 1**"

    <figure markdown>
    ![](https://www.mdpi.com/healthcare/healthcare-11-01946/article_deploy/html/images/healthcare-11-01946-g002.png){ alight=left width=800 }
    </figure>

Acquired resistance in bacteria involves modifying drug targets, drug inactivation, and drug efflux.
On the other hand, intrinsic resistance primarily stems from actions such as restricting drug uptake, drug inactivation, and drug efflux.
The structural differences between gram-positive and gram-negative bacteria contribute to variations in their drug resistance mechanisms.
Gram-positive bacteria tend to rely less on restricting drug uptake due to the absence of the lipopolysaccharide (LPS) outer membrane, and they have limited capabilities for certain types of drug efflux mechanisms.
In contrast, gram-negative bacteria have been observed to employ all four primary mechanisms of drug resistance.

### Restricting Drug Uptake

Gram-negative bacteria exhibit intrinsic resistance through lipopolysaccharide, a highly acylated glycolipid forming the major component of their outer membrane.
This structure acts as a permeability barrier, limiting the entry of various chemicals, including antibiotics.
This lowered permeability to specific antibiotics results in resistance.

Additionally, alterations in the permeability of outer-membrane proteins, particularly porin proteins, can lead to acquired resistance.
Porins are the primary entry points for hydrophilic antibiotics like β-lactams, fluoroquinolones, tetracyclines, and chloramphenicol.
The quantity and type of porin proteins influence the entry of these antibiotics into bacterial cells, impacting bacterial susceptibility.
Acquired resistance can also arise from mutations affecting porin expression alone or in conjunction with other resistance mechanisms like efflux pumps or enzymatic degradation of antibiotics.

Some bacteria, including *Enterococcus faecalis*, *Staphylococcus aureus*, *Staphylococcus epidermidis*, *Streptococcus viridans*, *E. coli*, *Klebsiella pneumoniae*, *Proteus mirabilis*, and *Pseudomonas aeruginosa*, employ the formation of biofilms as another method of antimicrobial resistance.
Biofilms, composed of microbial cells embedded in self-produced exopolysaccharides, obstruct antibiotic penetration and hinder the attainment of bactericidal concentrations over the entire biofilm.

### Modifying Drug Targets

Bacteria can develop resistance by altering the targets necessary for drug binding, making the drug unable to bind or bind poorly to the modified target.
This modification typically results from spontaneous gene mutations encoding proteins acting as drug targets.
For example, mutations affecting the quinolone-resistance-determining region (QRDR) in the DNA gyrase of both gram-positive and gram-negative bacteria lead to fluoroquinolone resistance.
Methylation is another efficient method of target modification, with examples including erm methylases against macrolides, lincosamides, and streptogramin B antibiotics in gram-positive and gram-negative bacteria.
Methylation of the cfr gene has also been associated with resistance development in various bacteria, such as *Proteus vulgaris*, *Staphylococcus spp.*, *Enterococcus spp.*, *Bacillus spp.*, and *E. coli*.
*Staphylococcus spp.* reduces its affinity to β-lactam antibiotics through an alternative penicillin-binding protein encoded by mecA and mecC genes.

### Inactivation of Drugs

Bacterial species may induce drug resistance by inactivating antibiotics through two main processes: degradation of the antibiotic or the transfer of a chemical group to the antibiotic.
β-lactamases, hydrolyzing enzymes produced by enterobacterales family members, are particularly effective in inactivating β-lactam antibiotics.
Initially known as penicillinases and cephalosporinases, these enzymes open the β-lactam ring structure, rendering the antibiotic ineffective in binding to the target penicillin-binding proteins.
Many enterobacterales family members and gram-positive bacteria, including *Staphylococcus aureus*, *Enterococcus faecalis*, and *Enterococcus faecium*, carry β-lactamase genes that are often transmitted via horizontal gene transfer (HGT).
Additionally, tetracycline can be hydrolyzed by an enzyme expressed by the tetX gene in certain bacteria.
The most common chemical groups for drug inactivation include acetyl, phosphoryl, and adenyl groups.
Phosphorylation and adenylation are predominant against aminoglycosides, while acetylation is diverse and used against aminoglycosides, chloramphenicol, streptogramins, and fluoroquinolones.

### Efflux of Drugs

Bacteria utilize energy-dependent efflux pumps on the cytoplasmic membrane to regulate the accumulation of antibacterial chemicals, including antibiotics, within bacterial cells.
These efflux pumps, such as the first plasmid-encoded pump in Escherichia coli identified in 1980, expel harmful compounds from the cell, allowing bacteria to control their internal environment.
Various gram-positive and gram-negative bacteria have diverse efflux mechanisms. Most efflux systems engage in multidrug efflux mechanisms encoded on the chromosome to ensure intrinsic drug resistance, while genes for substrate-specific efflux pumps are often found on mobile genetic elements.
There are six drug efflux pump families based on structure and energy source: ATP-binding cassette (ABC), major facilitator superfamily (MFS), multidrug and toxic compound extrusion (MATE), small multidrug resistance (SMR), resistance–nodulation–division (RND), and drug metabolite transporter (DMT) superfamily.
In gram-positive bacteria, the ABC and MFS families dominate and are carried on plasmids or encoded by chromosomal genes, while clinically significant efflux systems in gram-negative bacteria belong to the RND superfamily.

<!-- REFERENCES -->

[^salam2023antimicrobial]: Salam, M. A., Al-Amin, M. Y., Salam, M. T., Pawar, J. S., Akhter, N., Rabaan, A. A., & Alqumber, M. A. (2023, July). Antimicrobial resistance: a growing serious threat for global public health. In Healthcare (Vol. 11, No. 13, p. 1946). MDPI. doi: [10.3390/healthcare11131946](https://doi.org/10.3390/healthcare11131946)
[^pulingam2022antimicrobial]: Pulingam, T., Parumasivam, T., Gazzali, A. M., Sulaiman, A. M., Chee, J. Y., Lakshmanan, M., ... & Sudesh, K. (2022). Antimicrobial resistance: Prevalence, economic burden, mechanisms of resistance and strategies to overcome. *European Journal of Pharmaceutical Sciences, 170*, 106103. doi: [10.1016/j.ejps.2021.106103](https://doi.org/10.1016/j.ejps.2021.106103)
[^christoph2022good]: Crits-Christoph, A., Hallowell, H. A., Koutouvalis, K., & Suez, J. (2022). Good microbes, bad genes? The dissemination of antimicrobial resistance in the human microbiome. *Gut Microbes, 14*(1), 2055944. doi: [10.1080/19490976.2022.2055944](https://doi.org/10.1080/19490976.2022.2055944)
[^who2021global]: Global antimicrobial resistance and use surveillance system (GLASS) report 2021. Geneva: World Health Organization; 2021. [https://www.who.int/publications/i/item/9789240062702](https://www.who.int/publications/i/item/9789240062702)
[^samreen2021environmental]: Ahmad, I., Malak, H. A., & Abulreesh, H. H. (2021). Environmental antimicrobial resistance and its drivers: a potential threat to public health. *Journal of Global Antimicrobial Resistance, 27*, 101-111. doi: [10.1016/j.jgar.2021.08.001](https://doi.org/10.1016/j.jgar.2021.08.001)
[^zhang2021omics]: Zhang, A. N., Gaston, J. M., Dai, C. L., Zhao, S., Poyet, M., Groussin, M., ... & Zhang, T. (2021). An omics-based framework for assessing the health risk of antimicrobial resistance genes. *Nature communications, 12*(1), 4765. doi: [10.1038/s41467-021-25096-3](https://doi.org/10.1038/s41467-021-25096-3)
[^checcucci2020exploring]: Checcucci, A., Trevisi, P., Luise, D., Modesto, M., Blasioli, S., Braschi, I., & Mattarelli, P. (2020). Exploring the animal waste resistome: the spread of antimicrobial resistance genes through the use of livestock manure. *Frontiers in microbiology, 11*, 1416. doi: [10.3389/fmicb.2020.01416](https://doi.org/10.3389/fmicb.2020.01416)
