# Protein tertiary structure

The **tertiary structure** of a protein consists of the way a polypeptide is formed into complex three dimension shape. The tertiary structure will have a single polypeptide chain "backbone" with one or more protein secondary structures, such as **alpha helices**, **beta sheets**, or unstructured regions. Amino acid side chains may interact and bond in a number of ways. The interactions and bonds of side chains within a particular protein determine its tertiary structure. A number of tertiary structure subunits may fold into a **quaternary structure**.

## Determinants of tertiary structure

A protein folded into its native state or **native conformation** typically has a free energy than the unfolded conformation. A protein will tend towards low-energy conformations, which will determine the protein's fold in the cellular environment. Because many similar conformations will have similar energies, protein structures are dynamic, fluctuating between these similar structures.

**Globular proteins** have a core of hydrophobic amino acid residues and a surface region of water-exposed, charged, hydrophilic residues. This arrangement may stabilize interactions within the tertiary structure. For example, in secreted proteins, which are not bathed in cytoplasm, disulfide bonds between cysteine residues help to maintain the tertiary structure. The structure of a protein, such as an enzyme, depends on the cellular environment it is in, and may change upon binding of its natural ligands, for example a cofactor.

Proteins may be grouped and classified by the structures they hold. Databases of proteins which use such a classification include SCOP and CATH.



<!---There is a commonality of stable tertiary structures seen in proteins of diverse function and diverse evolution. F
or example, the TIM barrel, named for the enzyme triosephosphateisomerase, is a common tertiary structure as is the highly stable, dimeric, coiled coil structure. --->



<!---Kinetic traps
Folding kinetics may trap a protein in a high-energy conformation, i.e. a high-energy intermediate conformation blocks access to the lowest-energy conformation. The high-energy conformation may contribute to the function of the protein. For example, the influenza hemagglutinin protein is a single polypeptide chain which when activated, is proteolytically cleaved to form two polypeptide chains. The two chains are held in a high-energy conformation. When the local pH drops, the protein undergoes an energetically favorable conformational rearrangement that enables it to penetrate the host cell membrane.

Metastability
Some tertiary protein structures may exist in long-lived states that are not the expected most stable state. For example, many serpins (serine protease inhibitors) show this metastability. They undergo a conformational change when a loop of the protein is cut by a protease.[5][6][7]---->

<!---Chaperone proteins
It is commonly assumed that the native state of a protein is also the most thermodynamically stable and that a protein will reach its native state, given its chemical kinetics, before it is translated. Protein chaperones within the cytoplasm of a cell assist a newly synthesised polypeptide to attain its native state. Some chaperone proteins are highly specific in their function, for example, protein disulfide isomerase; others are general in their function and may assist most globular proteins, for example, the prokaryotic GroEL/GroES system of proteins and the homologous eukaryotic heat shock proteins (the Hsp60/Hsp10 system).---->

<!---### Cytoplasmic environment

Prediction of protein tertiary structure relies on knowing the protein's primary structure and comparing the possible predicted tertiary structure with known tertiary structures in protein data banks. This only takes into account the cytoplasmic environment present at the time of protein synthesis to the extent that a similar cytoplasmic environment may also have influenced the structure of the proteins recorded in the protein data bank.--->


<!---Structure stabilized by the formation of weak bonds between amino acid side chains - Determined by the folding of the polypeptide chain on itself (nonpolar residues are located inside the protein, while polar residues are mainly located outside) - Envelopment of the protein brings the protein closer and relates a-to located in distant regions of the sequence - Acquisition of the tertiary structure leads to the formation of pockets and sites suitable for the recognition and the binding of specific molecules (biospecificity).--->

## Determination of tertiary structure

Determining the structure of a protein is key to understanding its function and is a major task undertaken by *sStructural biologists**.  The knowledge of the tertiary structure of soluble globular proteins is more advanced than that of membrane proteins because the former are easier to study with available technology. Several advanced lab techniques are used to determine ("solve") protein structures.

### X-ray crystallography

<figure>
  <img src="https://upload.wikimedia.org/wikipedia/commons/0/0e/Protein_crystal.jpg" alt="A protein crystal" width="20%"/>
  <figcaption>A protein crystal seen under a microscope. Crystals used in X-ray crystallography may be smaller than a millimeter across.</figcaption>
</figure>


**X-ray crystallography** is by far the most common tool used to determine protein structure. It provides high resolution of the structure but it does not give information about protein's conformational flexibility. X-ray crystallography is the experimental science of empirically determining the atomic and molecular structure of a crystal, in which the crystalline structure causes a beam of X-rays to diffract into many specific directions. By measuring the angles and intensities of these diffracted beams, a crystallographer can produce a three-dimensional picture of the density of electrons within the crystal. From this electron density, the positions of the atoms in the crystal can be determined, as well as their chemical bonds.

<figure>
  <img src="https://upload.wikimedia.org/wikipedia/commons/7/7d/X-ray_diffraction_pattern_3clpro.jpg" alt="Diffraction pattern from a protein crystal" width="20%"/>
  <figcaption>An X-ray diffraction pattern of a crystallized enzyme. The pattern of spots (reflections) and the relative strength of each spot (intensities) can be used to determine the structure of the enzyme..</figcaption>
</figure>



Since many materials can form crystals—such as salts, metals, minerals, as well as various inorganic, organic, and biological molecules—X-ray crystallography has been fundamental in the development of many scientific fields. In its first decades of use, this method determined the size of atoms, the lengths and types of chemical bonds, and the atomic-scale differences among various materials. The method also revealed the structure and function of many biological molecules, including vitamins, drugs, proteins and nucleic acids such as DNA. X-ray crystallography is still the primary method for characterizing the atomic structure of new materials and in discerning materials that appear similar by other experiments.


<figure>
  <img src="https://upload.wikimedia.org/wikipedia/commons/6/60/Myoglobin.png" alt="Ribbon diagram of myoglobin" width="20%"/>
  <figcaption>Ribbon diagram of the structure of myoglobin, showing alpha helices. Such proteins are long, linear molecules with thousands of atoms; yet the relative position of each atom has been determined with sub-atomic resolution by X-ray crystallography. Since it is difficult to visualize all the atoms at once, the ribbon shows the rough path of the protein's backbone from its N-terminus to its C-terminus.1</figcaption>
</figure>



<!---In a single-crystal X-ray diffraction measurement, a crystal is mounted on a goniometer. The goniometer is used to position the crystal at selected orientations. The crystal is illuminated with a finely focused monochromatic beam of X-rays, producing a diffraction pattern of regularly spaced spots known as reflections. The two-dimensional images taken at different orientations are converted into a three-dimensional model of the density of electrons within the crystal using the mathematical method of Fourier transforms, combined with chemical data known for the sample. Poor resolution (fuzziness) or even errors may result if the crystals are too small, or not uniform enough in their internal makeup.--->

<!---X-ray crystallography is related to several other methods for determining atomic structures. Similar diffraction patterns can be produced by scattering electrons or neutrons. If single crystals of sufficient size cannot be obtained, various other X-ray methods can be applied to obtain less detailed information; such methods include fiber diffraction, powder diffraction and (if the sample is not crystallized) small-angle X-ray scattering (SAXS). If the material under investigation is only available in the form of nanocrystalline powders or suffers from poor crystallinity, the methods of electron diffraction, transmission electron microscopy and electron crystallography can be applied for determining the atomic structure.

For all above mentioned X-ray diffraction methods, the scattering is elastic; the scattered X-rays have the same wavelength as the incoming X-ray. By contrast, inelastic X-ray scattering methods are useful in studying excitations of the sample such as plasmons, crystal-field and orbital excitations, magnons, and phonons, rather than the distribution of its atoms.[1]--->

Crystal structures of proteins began to be solved in the late 1950s, beginning with the structure of sperm whale myoglobin by Sir John Cowdery Kendrew, for which he shared the Nobel Prize in Chemistry with Max Perutz in 1962. Since that success, over 130,000 X-ray crystal structures of proteins, nucleic acids and other biological molecules have been determined.

<!---X-ray crystallography is used routinely to determine how a pharmaceutical drug interacts with its protein target and what changes might improve it.[95] However, intrinsic membrane proteins remain challenging to crystallize because they require detergents or other denaturants to solubilize them in isolation, and such detergents often interfere with crystallization. Membrane proteins are a large component of the genome, and include many proteins of great physiological importance, such as ion channels and receptors.[96][97] Helium cryogenics are used to prevent radiation damage in protein crystals.[98]--->

<!---On the other end of the size scale, even relatively small molecules may pose challenges for the resolving power of X-ray crystallography. The structure assigned in 1991 to the antibiotic isolated from a marine organism, diazonamide A (C40H34Cl2N6O6, molar mass 765.65 g/mol), proved to be incorrect by the classical proof of structure: a synthetic sample was not identical to the natural product. The mistake was attributed to the inability of X-ray crystallography to distinguish between the correct -OH / -NH and the interchanged -NH2 / -O- groups in the incorrect structure.[99] With advances in instrumentation, however, similar groups can be distinguished using modern single-crystal X-ray diffractometers.--->

<!--- Despite being an invaluable tool in structural biology, protein crystallography carries some inherent problems in its methodology that hinder data interpretation. The crystal lattice, which is formed during the crystallization process, contains numerous units of the purified protein, which are densely and symmetrically packed in the crystal. When looking for a previously unknown protein, figuring out its shape and boundaries within the crystal lattice can be challenging. Proteins are usually composed of smaller subunits, and the task of distinguishing between the subunits and identifying the actual protein, can be challenging even for the experienced crystallographers. The non-biological interfaces that occur during crystallization are known as crystal-packing contacts (or simply, crystal contacts) and cannot be distinguished by crystallographic means. When a new protein structure is solved by X-ray crystallography and deposited in the Protein Data Bank, its authors are requested to specify the "biological assembly" which would constitute the functional, biologically-relevant protein. However, errors, missing data and inaccurate annotations during the submission of the data, give rise to obscure structures and compromise the reliability of the database. The error rate in the case of faulty annotations alone has been reported to be upwards of 6.6%[100] or approximately 15%,[101] arguably a non-trivial size considering the number of deposited structures. This "interface classification problem" is typically tackled by computational approaches and has become a recognized subject in structural bioinformatics.--->






<figure>
  <img src="https://upload.wikimedia.org/wikipedia/commons/thumb/7/73/X_ray_diffraction.png/1024px-X_ray_diffraction.png" alt="Ribbon diagram of myoglobin" width="20%"/>
  <figcaption>Workflow for solving the structure of a molecule by X-ray crystallography.</figcaption>
</figure>

### NMR

The nearest competing method to x-ray crystalography is nuclear magnetic resonance (NMR) spectroscopy, which has resolved less than one tenth as many macromolecules. **Protein Nuclear Magnetic Resonance (NMR)** imaging gives comparatively lower resolution of protein structure and is limited to smaller proteins. However, it can provide information about conformational changes of a protein in solution.

### Cryogenic electron microscopy

Cryogenic electron microscopy (cryo-EM) can give information about both a protein's tertiary and quaternary structure. It is particularly well-suited to large proteins and symmetrical complexes of protein subunits.

<!----Dual polarisation interferometry
Dual polarisation interferometry provides complementary information about surface captured proteins. It assists in determining structure and conformation changes over time.---->

## Molecular models of protein structures

Once a protein structure has been solved in the lab, the data can be used to create drawing, computer animations, 3D sculture, and other molecular models representing the data.  A **molecular model** is a physical, mathematical or other type of model of an atomistic system that represents molecules and their processes. They play an important role in understanding chemistry and generating and testing hypotheses. The creation of mathematical models of molecular properties and behavior is referred to as molecular modeling, and their graphical depiction is referred to as molecular graphics.

Molecular models may be created for several reasons – as teaching aids for students or those unfamiliar with atomistic structures; as objects to generate or test theories (e.g., the structure of DNA); as analogue computers (e.g., for measuring distances and angles in flexible systems); or as aesthetically pleasing objects on the boundary of art and science.

The construction of physical models is often a creative act, and many examples have been carefully created in the workshops of science departments. There is a very wide range of approaches to physical modeling, including ball-and-stick models available for purchase commercially, to molecular models created using 3D printers. Molecular graphics has made the visualization of molecular models on computer hardware easier, more accessible, and inexpensive, although physical models are widely used to enhance the tactile and visual message being portrayed.

Computers play a key role in building models because they can be used to predict bond lengths and angles, molecular polarity and charge distribution, and even quantum mechanical properties.  For most practical purposes, such as drug design or protein folding, the calculations of a model require supercomputing or cannot be done on classical computers at all in a reasonable amount of time.

### Ribbon diagrams

**Ribbon diagrams** are schematic representations of protein 3D structure and are one of the most common methods of protein depiction used today. The ribbon depicts the general course and organization of the protein backbone in 3D and serves as a visual framework for hanging details of the entire atomic structure, such as the balls for the oxygen atoms attached to myoglobin's active site in the adjacent figure. Ribbon diagrams are generated by interpolating a smooth curve through the polypeptide backbone. **α-helices** are shown as coiled ribbons or thick tubes, **β-sheets** as arrows, and non-repetitive coils or loops as lines or thin tubes. The direction of the polypeptide chain is shown locally by the arrows, and may be indicated overall by a colour ramp along the length of the ribbon.



Ribbon diagrams are simple yet powerful, expressing the visual basics of a molecular structure (twist, fold and unfold). This method has successfully portrayed the overall organization of protein structures, reflecting their three-dimensional nature and allowing better understanding of these complex objects both by expert structural biologists and by other scientists, students,[2] and the general public.


The first ribbon diagrams, hand-drawn by Jane Richardson in 1980 (influenced by earlier individual illustrations), were the first schematics of 3D protein structure to be produced systematically. In 1982, Arthur M. Lesk and co-workers first enabled automatic generation of ribbon diagrams through a computational implementation that uses **Protein Data Bank**  *.PDB files as input. Both hand-drawn and most computer ribbons are smoothed over to produce a more visually pleasing and understandable representation. Since their inception, and continuing in the present, ribbon diagrams have been the single most common representation of protein structure and a common choice of cover image for a journal or textbook.

#### Current computer programs for ribbon diagrams

Numerous programs exist for modeling proteins and rendering ribbon diagrams, including Molscript and UCSF Chimera.  **PyMOL** is a popular and flexible molecular graphics program (based on Python) that operates in interactive mode and also produces presentation-quality 2D images for ribbon diagrams and many other representations.

<figure>
  <img src="https://upload.wikimedia.org/wikipedia/commons/f/ff/Tubby-1c8z-pymol.png" alt="PyMol output" width="20%"/>
  <figcaption>Figure: PyMol ribbon of the structure of the tubby protein (PDB: 1C8Z​)</figcaption>
</figure>

The programs allow you to select the format, style, coloring and angle to best represent the protein and its structural features.

<figure>
  <img src="https://upload.wikimedia.org/wikipedia/commons/thumb/f/f2/Pymol_8_view.png/1599px-Pymol_8_view.png
" alt="PyMol output" width="70%"/>
  <figcaption>Figure: PyMol ribbon of the structure of the tubby protein (PDB: 1C8Z​)The same protein structure (TEV protease 1lvb) rendered in different modes. Standard cartoon, surface, cut-through of surface, highlighted barrels, 'QuteMol'-like, 'Goodsell'-like, b-factor putty



<!---Features
Spiral helix ribbons, beta-strand arrows, and smoothed loops, hand-drawn by Jane Richardson
Secondary structure[4][5]

α-Helices	Cylindrical spiral ribbons, with ribbon plane approximately following plane of peptides.
β-Strands	Arrows with thickness, about one-quarter as thick as they are wide, showing direction and twist of the strand from amino to carboxy end. β-sheets are seen as unified because neighboring strands twist in unison.
Loops and miscellaneous
Nonrepetitive loops	Round ropes that are fatter in the foreground and thinner towards the back, following smoothed path of Cα trace.
Junctions between loops and helices	Round rope that gradually flattens out into a thin helical ribbon.
Other features
Polypeptide direction,
NH2 and COOH termini

Small arrows on one or both of the termini, or letters. For β-strands, the direction of the arrow is sufficient. Today, the direction of the polypeptide chain is often indicated by a colour ramp.
Disulfide bonds	Interlocked SS symbol or a zigzag, like a stylized lightning stroke.
Prosthetic groups or inhibitors	Stick figures, or ball & stick.
Metals	Spheres.
Shading and colour	Shading or colour adds dimensionality to the diagram. Generally, the features at the front are the highest in contrast and those towards the back are the lowest.


## References

 Smith, Thomas J. (October 27, 2005). "Displaying and Analyzing Atomic Structures on the Macintosh". Danforth Plant Science Center. Archived from the original on 28 March 2002.
 Richardson, D. C.; Richardson, J. S. (January 2002). "Teaching Molecular 3-D Literacy". Biochemistry and Molecular Biology Education. 30 (1): 21–26. doi:10.1002/bmb.2002.494030010005.
 Richardson, Jane S. (2000), "Early ribbon drawings of proteins", Nature Structural Biology, 7 (8): 624–625, doi:10.1038/77912, PMID 10932243, S2CID 52856546.
 Richardson, Jane S. (1985), Schematic Drawings of Protein Structures, Methods in Enzymology, vol. 115, pp. 359–380, doi:10.1016/0076-6879(85)15026-3, ISBN 978-0-12-182015-2, PMID 3853075.
 Richardson, Jane S. (1981), Anatomy and Taxonomy of Protein Structures, Advances in Protein Chemistry, vol. 34, pp. 167–339, doi:10.1016/S0065-3233(08)60520-3, ISBN 978-0-12-034234-1, PMID 7020376.
 "Science's 'Mother of Ribbon Diagrams' celebrates 50 years at Duke". Duke Stories. 2018-10-19. Retrieved 2020-06-09.
 Lesk, Arthur M.; Hardman, Karl D. (1982), "Computer-Generated Schematic Diagrams of Protein Structures", Science, 216 (4545): 539–540, Bibcode:1982Sci...216..539L, doi:10.1126/science.7071602, PMID 7071602.
 Carson, M.; Bugg, C. E. (1986), "Algorithm for Ribbon Models of Proteins", Journal of Molecular Graphics, 4 (2): 121–122, doi:10.1016/0263-7855(86)80010-8.
 Richardson, D. C.; Richardson, J. S. (January 1992), "The kinemage: a tool for scientific communication", Protein Science, 1 (1): 3–9, doi:10.1002/pro.5560010102, PMC 2142077, PMID 1304880
 MolScript v2.1: About the program
 Chen, V. B.; Davis, I. W.; Richardson, D. C. (2009), "KING (Kinemage, Next Generation): A versatile interactive molecular and scientific visualization program", Protein Science, 18 (11): 2403–2409, doi:10.1002/pro.250, PMC 2788294, PMID 19768809
 Goddard, Thomas D.; Huang, Conrad C.; Ferrin, Thomas E. (2005), "Software Extensions to UCSF Chimera for Interactive Visualization of Large Molecular Assemblies", Structure, 13 (3): 473–482, doi:10.1016/j.str.2005.01.006, PMID 15766548.
 Brunger, Axel T.; Wells, James A. (2009), "Warren L. DeLano, 21 June 1972-3 November 2009", Nature Structural & Molecular Biology, 16 (12): 1202–1203, doi:10.1038/nsmb1209-1202, PMID 19956203.--->

### Space filling models

Another common way to represent protein tertiary structures is a **space filling model**.  In chemistry, a **space-filling model** is a type of three-dimensional (3D) molecular model where the atoms are represented by spheres whose radii are proportional to the radii of the atoms and whose center-to-center distances are proportional to the distances between the atomic nuclei, all in the same scale. Atoms of different chemical elements are usually represented by spheres of different colors.

An example of a space-filling model of a very complex macromolecule is shown below.  This is a protein, the cell membrane-spanning β2 adrenoreceptor, a G protein-coupled receptor, viewed as if looking down onto the extracellular surface. The electrostatic potential surface was applied to a model with atom positions determined by crystallography (PDB code 2RH1). It is shaded blue for electropositive areas to red for electronegative areas. Somewhat apparent, in stick representation in yellow, red and blue, in a groove at the top of the receptor, is a small molecule ligand bound to it. All of such binding interactions and the function of the receptor in signal transduction are mediated by electrostatic effects, and in modern structure work they are often studied using similar space filling models.

<figure>
  <img src="https://upload.wikimedia.org/wikipedia/commons/e/e8/Beta-2-adrenergic-receptor-electrostatic-top.png
" alt="Example space-filling model" width="20%"/>
  <figcaption>Figure: PyMol ribbon of the structure of the tubby protein (PDB: 1C8Z​)</figcaption>
</figure>



<!-- They are distinguished from other 3D representations, such as the ball-and-stick and skeletal models, by the use of the "full size" space-filling spheres for the atoms. The models are tactile and manually rotatable. They are useful for visualizing the effective shape and relative dimensions of a molecule, and (because of the rotatability) the shapes of the surface of the various conformers. On the other hand, these models mask the chemical bonds between the atoms, and make it difficult to see the structure of the molecule that is obscured by the atoms nearest to the viewer in a particular pose. For this reason, such models are of greater utility if they can be used dynamically, especially when used with complex molecules (e.g., see the greater understanding of the molecules shape given when the THC model is clicked on to rotate).-->

## Protein Structure Databases

After being solved the raw data for a protein's structure is deposited in databases to be available to the scientific community.  The **Protein Data Bank (PDB)** is the primary database for the three-dimensional structural data of large biological molecules, such as proteins and nucleic acids. The data are typically obtained by **X-ray crystallography**, NMR spectroscopy, or, increasingly, cryo-electron microscopy.  Structures are submitted by scientists from around the world, are freely accessible on the Internet via the websites of its member organisations.

The PDB is a key  resource in areas of **structural biology**, such as **structural genomics**. Most major scientific journals and some funding agencies now require scientists to submit their structure data to the PDB.  PDB data is the basis for the CASP competition and source data for the protein structure prediction software and database AlphaFold.

## Acknowledgements

This chapter was adapted from the following Wikipedia entries:

https://en.wikipedia.org/wiki/Protein_tertiary_structure
https://en.wikipedia.org/wiki/X-ray_crystallography
https://en.wikipedia.org/wiki/Molecular_model
https://en.wikipedia.org/wiki/Ribbon_diagram
https://en.wikipedia.org/wiki/PyMOL
https://en.wikipedia.org/wiki/Space-filling_model
https://en.wikipedia.org/wiki/Protein_Data_Bank


<!-- ## History

Two forces converged to initiate the PDB: a small but growing collection of sets of protein structure data determined by X-ray diffraction; and the newly available (1968) molecular graphics display, the Brookhaven RAster Display (BRAD), to visualize these protein structures in 3-D. In 1969, with the sponsorship of Walter Hamilton at the Brookhaven National Laboratory, Edgar Meyer (Texas A&M University) began to write software to store atomic coordinate files in a common format to make them available for geometric and graphical evaluation. By 1971, one of Meyer's programs, SEARCH, enabled researchers to remotely access information from the database to study protein structures offline.[8] SEARCH was instrumental in enabling networking, thus marking the functional beginning of the PDB.

The Protein Data Bank was announced in October 1971 in Nature New Biology[9] as a joint venture between Cambridge Crystallographic Data Centre, UK and Brookhaven National Laboratory, US.

Upon Hamilton's death in 1973, Tom Koeztle took over direction of the PDB for the subsequent 20 years. In January 1994, Joel Sussman of Israel's Weizmann Institute of Science was appointed head of the PDB. In October 1998,[10] the PDB was transferred to the Research Collaboratory for Structural Bioinformatics (RCSB);[11] the transfer was completed in June 1999. The new director was Helen M. Berman of Rutgers University (one of the managing institutions of the RCSB, the other being the San Diego Supercomputer Center at UC San Diego).[12] In 2003, with the formation of the wwPDB, the PDB became an international organization. The founding members are PDBe (Europe),[2] RCSB (US), and PDBj (Japan).[3] The BMRB[5] joined in 2006. Each of the four members of wwPDB can act as deposition, data processing and distribution centers for PDB data. The data processing refers to the fact that wwPDB staff review and annotate each submitted entry.[13] The data are then automatically checked for plausibility (the source code[14] for this validation software has been made available to the public at no charge).--->

<!---Contents

Examples of protein structures from the PDB (created with UCSF Chimera)

Rate of Protein Structure Determination by Method and Year. MX = macromolecular crystallography, 3DEM = 3D Electron Microscopy.[15]
The PDB database is updated weekly (UTC+0 Wednesday), along with its holdings list.[16] As of 10 January 2023, the PDB comprised:

Experimental
Method	Proteins only	Proteins with oligosaccharides	Protein/Nucleic Acid
complexes	Nucleic Acids only	Other	Oligosaccharides only	Total
X-ray diffraction	152277	8969	8027	2566	163	11	172013
NMR	12104	32	281	1433	31	6	13887
Electron microscopy	9226	1633	2898	77	8	0	13842
Hybrid	189	7	6	12	0	1	215
Neutron	72	1	0	2	0	0	75
Other	32	0	0	1	0	4	309
Total:	173900	10642	11212	4091	202	22	200069
162,041 structures in the PDB have a structure factor file.
11,242 structures have an NMR restraint file.
5,774 structures in the PDB have a chemical shifts file.
13,388 structures in the PDB have a 3DEM map file deposited in EM Data Bank
Most structures are determined by X-ray diffraction, but about 7% of structures are determined by protein NMR. When using X-ray diffraction, approximations of the coordinates of the atoms of the protein are obtained, whereas using NMR, the distance between pairs of atoms of the protein is estimated. The final conformation of the protein is obtained from NMR by solving a distance geometry problem. After 2013, a growing number of proteins are determined by cryo-electron microscopy.

For PDB structures determined by X-ray diffraction that have a structure factor file, their electron density map may be viewed. The data of such structures may be viewed on the three PDB websites.

Historically, the number of structures in the PDB has grown at an approximately exponential rate, with 100 registered structures in 1982, 1,000 structures in 1993, 10,000 in 1999, 100,000 in 2014, and 200,000 in January 2023.[17][18]

File format
Main article: Protein Data Bank (file format)
The file format initially used by the PDB was called the PDB file format. The original format was restricted by the width of computer punch cards to 80 characters per line. Around 1996, the "macromolecular Crystallographic Information file" format, mmCIF, which is an extension of the CIF format was phased in. mmCIF became the standard format for the PDB archive in 2014.[19] In 2019, the wwPDB announced that depositions for crystallographic methods would only be accepted in mmCIF format.[20]

An XML version of PDB, called PDBML, was described in 2005.[21] The structure files can be downloaded in any of these three formats, though an increasing number of structures do not fit the legacy PDB format. Individual files are easily downloaded into graphics packages from Internet URLs:

For PDB format files, use, e.g., http://www.pdb.org/pdb/files/4hhb.pdb.gz or http://pdbe.org/download/4hhb
For PDBML (XML) files, use, e.g., http://www.pdb.org/pdb/files/4hhb.xml.gz or http://pdbe.org/pdbml/4hhb
The "4hhb" is the PDB identifier. Each structure published in PDB receives a four-character alphanumeric identifier, its PDB ID. (This is not a unique identifier for biomolecules, because several structures for the same molecule—in different environments or conformations—may be contained in PDB with different PDB IDs.)

Viewing the data
The structure files may be viewed using one of several free and open source computer programs, including Jmol, Pymol, VMD, Molstar and Rasmol. Other non-free, shareware programs include ICM-Browser,[22] MDL Chime, UCSF Chimera, Swiss-PDB Viewer,[23] StarBiochem[24] (a Java-based interactive molecular viewer with integrated search of protein databank), Sirius, and VisProt3DS[25] (a tool for Protein Visualization in 3D stereoscopic view in anaglyph and other modes), and Discovery Studio. The RCSB PDB website contains an extensive list of both free and commercial molecule visualization programs and web browser plugins.--->
