# De novo

TODO:

## Tools

| Name | Main feature |
| ---- | ------------ |
| [ABySS2](https://github.com/bcgsc/abyss) | An assembler intended for SGS PE and linked reads. |
| [AllPath-LG](https://www.msi.umn.edu/sw/allpaths-lg) | Uses a unipath graph from the k-mer paths to collapse repeats. |
| [MEGAHIT](https://github.com/voutcn/megahit) | An ultrafast and memory-efficient assembler for SGS reads. |
| [SOAPdenovo](https://github.com/aquaskyline/SOAPdenovo2) | De Bruijn graph assembler with an error correction stage. |
| [Falcon/HGAP](https://pb-falcon.readthedocs.io/en/latest/about.html) | Diploid-aware mode including trim, correction, and consensus for PacBio reads. |
| [CANU](https://canu.readthedocs.io/en/latest/) | A fork of the Celera Assembler, including trim, correction, and consensus for TGS reads. |
| [SMARTdenovo](https://github.com/ruanjue/smartdenovo) | De novo assembler including all-vs.-all raw read alignments without an error correction stage for TGS reads. |
| [MECAT](https://github.com/xiaochuanle/MECAT2) | Ultrafast mapping, error correction, and de novo assembly tools for single-molecule sequencing reads. |
| [Flye](https://github.com/fenderglass/Flye) | A repeat graph mode including trim, correction, and consensus with polishing for TGS reads. |
| [Shasta](https://github.com/paoloshasta/shasta) | A run-length representation of ONT reads. |
